node-agent/workload_manager/src/workload_manager/workload/mod.rs

```diff
@@ -0,0 +1,17 @@
+use self::errors::WorkloadError;
+use proto::agent::Instance;
+use workload_trait::Workload;
+
+mod container;
+pub mod errors;
+pub mod workload_trait;
+
+pub async fn create(instance: Instance) -> Result<impl Workload, WorkloadError> {
+    match instance.r#type {
+        // 0 for Container
+        0 => Ok(container::Container::new(instance).await),
```

Comment:
I'm hoping there's an enum for the instance type?

Reply:
Yes, sorry...